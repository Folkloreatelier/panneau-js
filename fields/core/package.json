{
    "name": "@panneau/field",
    "version": "0.6.30",
    "description": "Field core for Panneau",
    "keywords": [
        "react",
        "javascript",
        "form",
        "dashboard"
    ],
    "homepage": "https://github.com/Folkloreatelier/panneau-js/tree/master/fields/core",
    "repository": {
        "type": "git",
        "url": "git+https://github.com/Folkloreatelier/panneau-js.git"
    },
    "author": {
        "name": "Folklore",
        "email": "info@atelierfolklore.ca"
    },
    "contributors": [
        {
            "name": "David Mongeau-Petitpas",
            "email": "dmp@atelierfolklore.ca"
        },
        {
            "name": "Nicolas Roy-Bourdages",
            "email": "nrb@atelierfolklore.ca"
        }
    ],
    "main": "lib/index.js",
    "module": "es/index.js",
    "files": [
        "lib",
        "es",
        "dist"
    ],
    "scripts": {
        "storybook": "../../node_modules/.bin/start-storybook -p 9008 -c ../../.storybook-package",
        "clean": "rm -rf dist && rm -rf lib && rm -rf es",
        "webpack:dist": "../../node_modules/.bin/webpack --config ./webpack.config.js --progress --colors",
        "webpack": "npm run webpack:dist",
        "babel:es": "BABEL_ENV=es ../../node_modules/.bin/babel src --out-dir es --ignore src/**/*.story.jsx,src/**/*.test.jsx",
        "babel:cjs": "BABEL_ENV=cjs ../../node_modules/.bin/babel src --out-dir lib --ignore src/**/*.story.jsx,src/**/*.test.jsx",
        "babel": "npm run babel:es && npm run babel:cjs",
        "intl": "node ../../build/buildPackageTranslations",
        "build": "npm run babel && npm run intl",
        "prepare": "npm run clean && npm run build",
        "start": "npm run storybook"
    },
    "dependencies": {
        "@babel/runtime": "^7.0.0",
        "@panneau/core": "^0.6.30",
        "classnames": "^2.2.5",
        "fine-uploader": "^5.16.0",
        "fine-uploader-wrappers": "^1.0.1",
        "lodash": "^4.17.4",
        "moment": "^2.21.0",
        "prop-types": "^15.6.0",
        "react-fine-uploader": "^1.0.9",
        "react-intl": "^2.4.0",
        "react-loadable": "^5.5.0",
        "react-markdown": "^3.1.4"
    },
    "devDependencies": {
        "bootstrap": "~4.1.1",
        "react": "^16.7.0",
        "react-dom": "^16.7.0"
    },
    "peerDependencies": {
        "react": ">=15.0.0 || ^16.0.0",
        "react-dom": ">=15.0.0 || ^16.0.0"
    },
    "gitHead": "ec6b2a4882879b80298705db094c6c739e1802ae"
}
