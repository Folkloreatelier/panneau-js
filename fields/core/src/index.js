import FormGroup from './FormGroup';
import FieldsGroup from './FieldsGroup';
import Card from './Card';

export {
    FormGroup,
    FieldsGroup,
    Card,
};
