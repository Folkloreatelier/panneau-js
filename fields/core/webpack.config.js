/* eslint-disable import/no-extraneous-dependencies */
const webpackMerge = require('webpack-merge');
const webpackConfig = require('../../build/webpack.config.dist');
/* eslint-enable import/no-extraneous-dependencies */

module.exports = env => (
    webpackMerge(webpackConfig(env), {
        entry: {
            'panneau-form-group': './index',
        },
        output: {
            library: 'PanneauFormGroup',
        },
    })
);
