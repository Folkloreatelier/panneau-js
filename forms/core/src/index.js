import * as PropTypes from './PropTypes';
import withForm from './withForm';

export { PropTypes, withForm };
