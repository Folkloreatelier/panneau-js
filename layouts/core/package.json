{
  "name": "@panneau/layout",
  "version": "1.0.0-alpha.0",
  "description": "Layout core for Panneau",
  "keywords": [
    "react",
    "javascript",
    "form",
    "dashboard"
  ],
  "homepage": "https://github.com/folkloreinc/panneau-js/tree/master/layouts/core",
  "repository": {
    "type": "git",
    "url": "git+https://github.com/folkloreinc/panneau-js.git"
  },
  "author": {
    "name": "Folklore",
    "email": "info@folklore.email"
  },
  "contributors": [
    {
      "name": "David Mongeau-Petitpas",
      "email": "dmp@folklore.email"
    },
    {
      "name": "Nicolas Roy-Bourdages",
      "email": "nrb@folklore.email"
    }
  ],
  "main": "lib/index.js",
  "module": "es/index.js",
  "files": [
    "lib",
    "es",
    "dist"
  ],
  "scripts": {
    "storybook": "../../node_modules/.bin/start-storybook -p 9008 -c ../../.storybook-package",
    "clean": "rm -rf dist && rm -rf lib && rm -rf es",
    "webpack:dist": "../../node_modules/.bin/webpack --config ./webpack.config.js --progress --colors",
    "webpack": "npm run webpack:dist",
    "babel:es": "BABEL_ENV=es ../../node_modules/.bin/babel src --out-dir es --ignore src/**/*.story.jsx,src/**/*.test.jsx",
    "babel:cjs": "BABEL_ENV=cjs ../../node_modules/.bin/babel src --out-dir lib --ignore src/**/*.story.jsx,src/**/*.test.jsx",
    "babel": "npm run babel:es && npm run babel:cjs",
    "intl": "node ../../scripts/buildPackageTranslations",
    "build": "npm run babel && npm run intl",
    "prepare": "npm run clean && npm run build",
    "start": "npm run storybook"
  },
  "dependencies": {
    "@babel/runtime": "^7.5.5",
    "@panneau/core": "^1.0.0-alpha.0",
    "@panneau/field": "^1.0.0-alpha.0",
    "classnames": "^2.2.5",
    "lodash": "^4.17.4",
    "pascal-case": "^2.0.1",
    "prop-types": "^15.7.0",
    "react-intl": "^3.1.6",
    "react-router-dom": "^5.0.1"
  },
  "devDependencies": {
    "bootstrap": "^4.3.1",
    "react": "^16.9.0",
    "react-dom": "^16.9.0",
    "react-router": "^5.0.1"
  },
  "peerDependencies": {
    "react": "^16.8.0",
    "react-dom": "^16.8.0"
  },
  "gitHead": "8947c52802d8c1596eff0d4edd9d1447c1b2d020"
}
