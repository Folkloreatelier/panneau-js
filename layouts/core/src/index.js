import Header from './Header';
import Footer from './Footer';
import navbarItemsCollection from './navbarItems';

export {
    Header,
    Footer,
    navbarItemsCollection,
};
