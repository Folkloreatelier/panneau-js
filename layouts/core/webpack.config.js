/* eslint-disable import/no-extraneous-dependencies */
const path = require('path');
const webpackMerge = require('webpack-merge');
const webpackConfig = require('../../build/webpack.config.prod');

module.exports = () => (
    webpackMerge(webpackConfig, {
        entry: {
            'panneau-layout': [
                path.join(__dirname, '../../build/polyfills'),
                path.join(__dirname, './src/index'),
            ],
        },
        output: {
            library: 'PanneauLayout',
        },
    })
);
