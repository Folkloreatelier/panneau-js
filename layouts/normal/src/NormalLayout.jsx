import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Header, Footer } from '@panneau/layout';
import get from 'lodash/get';
import { connect } from 'react-redux';

import styles from './styles.scss';

const propTypes = {
    children: PropTypes.node,
    applicationDefinition: PropTypes.shape({
        name: PropTypes.string,
    }),
    definition: PropTypes.shape({
        header: PropTypes.oneOfType([
            PropTypes.bool,
            PropTypes.object,
        ]),
        footer: PropTypes.oneOfType([
            PropTypes.bool,
            PropTypes.object,
        ]),
    }),
    gotoHome: PropTypes.func,
    goto: PropTypes.func,
};

const defaultProps = {
    children: null,
    applicationDefinition: null,
    definition: null,
    gotoHome: null,
    goto: null,
};

class NormalLayout extends Component {
    constructor(props) {
        super(props);

        this.onHeaderClickMenuItem = this.onHeaderClickMenuItem.bind(this);
    }

    onHeaderClickMenuItem() {

    }

    render() {
        const {
            children,
            applicationDefinition,
            definition,
            gotoHome,
            goto,
        } = this.props;

        const title = get(applicationDefinition, 'name', 'Panneau');
        const header = get(definition, 'header', true);
        const footer = get(definition, 'footer', true);

        return (
            <div className={styles.container}>
                { header !== false ? (
                    <div className={styles.header}>
                        <Header
                            title={title}
                            {...header}
                            gotoHome={gotoHome}
                            goto={goto}
                            onClickMenuItem={this.onHeaderClickMenuItem}
                        />
                    </div>
                ) : null }

                <div className={styles.content}>
                    { children }
                </div>

                { footer !== false ? (
                    <div className={styles.footer}>
                        <Footer
                            {...footer}
                        />
                    </div>
                ) : null }
            </div>
        );
    }
}

NormalLayout.propTypes = propTypes;
NormalLayout.defaultProps = defaultProps;

const mapStateToProps = () => ({

});

const mapDispatchToProps = () => ({

});

const WithStoreContainer = connect(mapStateToProps, mapDispatchToProps)(NormalLayout);
export default WithStoreContainer;
