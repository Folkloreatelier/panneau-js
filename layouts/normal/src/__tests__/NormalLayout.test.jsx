import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import { Provider } from 'react-redux';
import { createStore } from '@panneau/core';

import NormalLayout from '../NormalLayout';

const store = createStore();

test('match snapshot', () => {
    const component = renderer.create((
        <Provider store={store}>
            <NormalLayout

            />
        </Provider>
    ));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
