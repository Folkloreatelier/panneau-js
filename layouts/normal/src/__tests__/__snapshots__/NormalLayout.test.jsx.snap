// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className={undefined}
    >
      <div
        className={undefined}
      >
        <nav
          className="navbar navbar-default"
        >
          <div
            className="container-fluid"
          >
            <div
              className="navbar-header"
            >
              <button
                aria-expanded="false"
                className="navbar-toggle collapsed"
                onClick={[Function]}
                type="button"
              >
                <span
                  className="sr-only"
                >
                  Menu
                </span>
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
                <span
                  className="icon-bar"
                />
              </button>
              <a
                className="navbar-brand"
                href="#"
                onClick={[Function]}
              >
                Panneau
              </a>
            </div>
            <div
              className="collapse navbar-collapse"
            >
              <ul
                className="nav navbar-nav"
              />
              <ul
                className="nav navbar-nav navbar-right"
              />
            </div>
          </div>
        </nav>
      </div>
    </div>
  </div>
  <div
    className={undefined}
  />
  <div
    className={undefined}
  >
    <div />
  </div>
</div>
`;
